cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -469,6 +475,187 @@ class self_comparator {
   std::shared_ptr<preprocessed_table> d_t;
 };
 
+enum class lhs_index_type : size_type {};
+enum class rhs_index_type : size_type {};
+
+struct make_lhs_index {
+  __device__ lhs_index_type operator()(size_type i) const { return static_cast<lhs_index_type>(i); }
+};
+
+struct make_rhs_index {
+  __device__ rhs_index_type operator()(size_type i) const { return static_cast<rhs_index_type>(i); }
+};
+
+auto inline make_lhs_index_counting_iterator(size_type start)
+{
+  return cudf::detail::make_counting_transform_iterator(start, make_lhs_index{});
+};
+
+auto inline make_rhs_index_counting_iterator(size_type start)
+{
+  return cudf::detail::make_counting_transform_iterator(start, make_rhs_index{});
+};
```

Comment:
Wait, I just realized that strong typed index is also relevant for equality comparator. So please move these utilities into the top of the file, in `cudf::experiment::` namespace.

Reply:
Good idea. Done.